diff --git a/src/AppBundle/Form/DataTransformer/IssueToNumberTransformer.php b/src/AppBundle/Form/DataTransformer/IssueToNumberTransformer.php
new file mode 100644
index 0000000..ff1c01e
--- /dev/null
+++ b/src/AppBundle/Form/DataTransformer/IssueToNumberTransformer.php
@@ -0,0 +1,72 @@
+<?php
+
+namespace AppBundle\Form\DataTransformer;
+
+use AppBundle\Entity\Issue;
+use Doctrine\Common\Persistence\ObjectManager;
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+class IssueToNumberTransformer implements DataTransformerInterface
+{
+    /**
+     * @var ObjectManager
+     */
+    private $em;
+
+    /**
+     * @param ObjectManager $em
+     */
+    public function __construct(ObjectManager $em)
+    {
+        $this->em = $em;
+    }
+
+    /**
+     * Transforms an object (issue) to a string (number).
+     *
+     * @param  Issue|null $issue
+     * @return string
+     */
+    public function transform($issue)
+    {
+        if (null === $issue) {
+            return '';
+        }
+
+        return $issue->getId();
+    }
+
+    /**
+     * Transforms a string (number) to an object (issue).
+     *
+     * @param  string $issueNumber
+     * @return Issue|null
+     * @throws TransformationFailedException if object (issue) is not found.
+     */
+    public function reverseTransform($issueNumber)
+    {
+        // no issue number? It's optional, so that's ok
+        if (!$issueNumber) {
+            return null;
+        }
+
+        $issue = $this->em
+            ->getRepository('AppBundle:Issue')
+            // query for the issue with this id
+            ->find($issueNumber)
+        ;
+
+        if (null === $issue) {
+            // causes a validation error
+            // this message is not shown to the user
+            // see the invalid_message option
+            throw new TransformationFailedException(sprintf(
+                'An issue with number "%s" does not exist!',
+                $issueNumber
+            ));
+        }
+
+        return $issue;
+    }
+}
