diff --git a/src/AppBundle/Form/TaskType.php b/src/AppBundle/Form/TaskType.php
index a5d6028..15eb33d 100644
--- a/src/AppBundle/Form/TaskType.php
+++ b/src/AppBundle/Form/TaskType.php
@@ -22,7 +22,10 @@ class TaskType extends AbstractType
     {
         $builder
             ->add('description', 'textarea')
-            ->add('issue', 'text');
+            ->add('issue', 'text', array(
+                // validation message if the data transformer fails
+                'invalid_message' => 'That is not a valid issue number'
+            ));
 
         $builder->get('description')
             ->addModelTransformer(new CallbackTransformer(
