diff --git a/src/AppBundle/Form/TaskType.php b/src/AppBundle/Form/TaskType.php
index 0daabaa..78f9e45 100644
--- a/src/AppBundle/Form/TaskType.php
+++ b/src/AppBundle/Form/TaskType.php
@@ -3,6 +3,7 @@
 namespace AppBundle\Form;
 
 use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\CallbackTransformer;
 use Symfony\Component\Form\FormBuilderInterface;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
@@ -11,11 +12,26 @@ class TaskType extends AbstractType
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $builder
-            ->add('description', 'text')
+            ->add('description', 'textarea')
             ->add('issue', 'entity', array(
                 'class' => 'AppBundle\Entity\Issue',
                 'property' => 'id'
             ));
+
+        $builder->get('description')
+            ->addModelTransformer(new CallbackTransformer(
+                // transform <br/> to \n so the textarea reads easier
+                function ($originalDescription) {
+                    return preg_replace('#<br\s*/?>#i', "\n", $originalDescription);
+                },
+                function ($submittedDescription) {
+                    // remove most HTML tags (but not br,p)
+                    $cleaned = strip_tags($submittedDescription, '<br><br/><p>');
+
+                    // transform any \n to real <br/>
+                    return str_replace("\n", '<br/>', $cleaned);
+                }
+            ));
     }
 
     public function configureOptions(OptionsResolver $resolver)
